plutus-tx-plugin/test/Inline/Spec.hs

```diff
@@ -98,6 +102,17 @@ factorial x
   | otherwise = x `PlutusTx.multiplyInteger` factorial (x `PlutusTx.subtractInteger` 1)
 {-# INLINEABLE factorial #-}
 
+-- | This test case verifies that `inline` can inline local bindings
+-- (like `square`).
+--
+-- The third usage of `square` is inlined in PIR, but not in UPLC, since
+-- in UPLC the inlining is reversed by CSE.
```

Comment:
It seems to me that this test should test inlining behavior in isolation from CSE, for which we can turn off it using the compiler flag.

If you want to demonstrate an interplay between CSE/inlining - this could be done in a dedicated unit test.

Reply:
The pir already proves that the inlining works. Turning off CSE proves nothing further. Instead, this way, it shows that both inlining and CSE works as expected - two birds with one stone.

Reply:
I initially missed that part too, but now I see that @zliu41 is right.